dvc/repo/__init__.py

```diff
@@ -122,19 +127,21 @@ def init(root_dir=os.curdir, no_scm=False, force=False):
     def unprotect(self, target):
         return self.cache.local.unprotect(PathInfo(target))
 
-    def _ignore(self):
+    def _ignore(self, no_state=False):
         from dvc.updater import Updater
 
         updater = Updater(self.dvc_dir)
 
         flist = [
-            self.state.state_file,
             self.lock.lock_file,
             self.config.config_local_file,
             updater.updater_file,
             updater.lock.lock_file,
         ] + self.state.temp_files
 
+        if not no_state:
+            flist += [self.state.state_file]
```

Comment:
It is a little weird that we have temp_files, but for state_file we have a special "if". How about we define `state_files = []` and use it instead of `state_file` same as we do with `temp_files`?